src/pretix/presale/views/event.py

```diff
@@ -104,6 +108,29 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class EventIcalDownload(EventViewMixin, View):
+    def get(self, request, *args, **kwargs):
+        if not self.request.event:
+            raise Http404(_('Unknown event code or not authorized to access this event.'))
+
+        cal = Calendar()
+        cal.add('version', '2.0')
+        cal.add('prodid', '-//{}//{}//'.format(self.request.event.organizer.name, self.request.event.name))
+
+        event = Event()
+        event.add('summary', self.request.event.name)
+        event.add('dtstart', self.request.event.date_from)
```

Comment:
This will output the date in the wrong format, you will need to call ``strftime`` to bring it in the correct format. If you have trouble doing that, I can look up the correct arguments for you, it might be non-trivial with regard to timezones ;) This of course is valid for all date properties.

The other thing is that there are two settings that we should respect here:

* ``self.request.event.settings.show_times`` -- If this is ``False``, we should not show the time of day, so the icalendar should contain a full-day event. Can you look up how to do this?
* ``self.request.event.settings.show_date_to`` -- If this is ``False`` we should only show the beginning date. As the icalendar needs a end date, I do not see a good way to respect that currently and I think we can ignore it here.